### ISSUE TITLE
Windows access violation in PuLIDFlux2 injector init when base is GGUF Flux 2 Dev on 32GB RAM systems (working diff inside)

### ISSUE THREAD

**Issue by Booyaka101** [iFayens/ComfyUI-PuLID-Flux2#19]
Filing as an issue with the working diff inline rather than a cold PR — would like to gauge whether this approach makes sense before opening one, since I have only verified on a single hardware config (RTX 4090 24GB / 32GB DDR5 / Win11 / py3.12.8 / torch 2.9.1+cu128).

### The crash

When loading Flux 2 Dev as GGUF (via `city96/ComfyUI-GGUF`, e.g. `flux2-dev-Q5_K_S.gguf`), then applying PuLID via `ApplyPuLIDFlux2`, Comfy hard-crashes during sampling with a Windows fatal exception (access violation):

```
[PuLID] 🔄 Projection 4096 → 6144 (flux2_dev)
🟡 PuLID Flux2 | FLUX2_DEV | strength=0.60 | face=0
Windows fatal exception: access violation

Stack (most recent call first):
  File ".../torch/nn/modules/linear.py", line 109 in __init__
  File ".../ComfyUI-PuLID-Flux2/pulid_flux2.py", line 59 in __init__
  File ".../ComfyUI-PuLID-Flux2/pulid_flux2.py", line 120 in __init__
  File ".../ComfyUI-PuLID-Flux2/pulid_flux2.py", line 453 in apply
```

The Comfy server process dies entirely — subsequent requests hit ECONNREFUSED.

### Root cause

Line 453 builds the injector for Flux 2 Dev:

```python
injector = PuLIDFlux2(dim=flux_dim).to(device, dtype=dtype)
```

With dim=6144 (Flux 2 Dev hidden), `PuLIDFlux2.__init__` constructs:
- `IDFormer` (1 PerceiverAttentionCA + 4 inner) → ~50M params
- `double_ca` (12 PerceiverAttentionCA) → ~300M params
- `single_ca` (60 PerceiverAttentionCA) → ~1.5B params

Total ~1.85B params. Default PyTorch `nn.Module` construction allocates these as fp32 weights on CPU first (~7-8GB), then `.to(device, dtype=dtype)` casts and moves to GPU.

Under GGUF Flux 2 Dev, the base model is being lazily dequantized in CPU RAM during inference patches (per `move_patch_to_device` in `ComfyUI-GGUF/ops.py`), which competes for the same CPU memory pool. On a 32GB-RAM box, the simultaneous ~16GB of allocations triggers a Windows OS-level allocation failure that bubbles up as an access violation in the `nn.Linear` constructor rather than a clean OOM.

The crash does NOT happen with `flux2_dev_fp8mixed.safetensors` because that path keeps the base model on disk via mmap and doesnt use the same CPU-side dequant pool.

### Fix (meta-device init)

Skip the CPU allocation entirely. Construct the module structure on the `meta` device (no real memory), then materialize directly on GPU via `to_empty`, then re-init weights in-place. Since the dim=6144 injector at line 453 is constructed with random init anyway (the Klein PuLID checkpoint is dim=4096 — line 453 is just a shape-compat hack, not loaded weights), re-initing on GPU has zero quality impact.

```diff
--- a/pulid_flux2.py
+++ b/pulid_flux2.py
@@ -5,6 +5,7 @@ Version optimisée avec adaptation dynamique à chaque run
 """

 import os
+import gc
 import torch
 import torch.nn as nn
 import torch.nn.functional as F
@@ -448,9 +449,30 @@ class ApplyPuLIDFlux2:
             nn.init.normal_(proj.weight, std=0.02)
             id_tokens = proj(id_tokens)
             id_tokens = F.normalize(id_tokens, p=2, dim=-1)
-
-            # Création d'un injector avec les bonnes dimensions pour Dev
-            injector = PuLIDFlux2(dim=flux_dim).to(device, dtype=dtype)
+
+            # Skip CPU fp32 staging - it competes with GGUF base-model dequant
+            # on 32GB-RAM systems and triggers Windows access violation in
+            # nn.Linear init. Meta-device construction skips CPU entirely;
+            # weights are random-init anyway so re-init on GPU is equivalent.
+            torch.cuda.empty_cache()
+            torch.cuda.synchronize()
+            gc.collect()
+            with torch.device("meta"):
+                injector = PuLIDFlux2(dim=flux_dim)
+            injector = injector.to_empty(device=device).to(dtype=dtype)
+            for _m in injector.modules():
+                if isinstance(_m, nn.Linear):
+                    nn.init.kaiming_uniform_(_m.weight, a=5 ** 0.5)
+                    if _m.bias is not None:
+                        nn.init.zeros_(_m.bias)
+                elif isinstance(_m, nn.LayerNorm):
+                    nn.init.ones_(_m.weight)
+                    nn.init.zeros_(_m.bias)
+            # latents Parameter inside IDFormer needs explicit re-init too
+            for _name, _p in injector.named_parameters():
+                if _name.endswith(".latents"):
+                    nn.init.normal_(_p, std=0.02)
         else:
             injector = pulid_model
```

### Verification

- Win11, RTX 4090 24GB, 32GB DDR5, py3.12.8, torch 2.9.1+cu128
- ComfyUI 0.19.3, ComfyUI-GGUF (`city96`) loading `flux2-dev-Q5_K_S.gguf` (22GB)
- Workflow: GGUF Flux 2 Dev base + Flux2 Turbo LoRA + ApplyPuLIDFlux2 + Flux2FunControlNet
- Before patch: Windows access violation on first `apply()`, Comfy server dies
- After patch: PuLID applies cleanly (`🟡 PuLID Flux2 | FLUX2_DEV | strength=0.60 | face=0`), sampling proceeds

### Caveat / what would help confirm before turning this into a PR

I have only verified on the single config above. Would help to know if:
- 64GB+ RAM systems hit the same crash (predict: no, since CPU has more headroom)
- Linux systems hit it (predict: maybe, depends on glibc malloc behavior under the same pressure)
- Q4_K_S or Q6_K GGUF variants change the threshold (predict: same crash, slightly different memory pressure)
- Klein 9B PuLID at dim=4096 also benefits (probably negligible since allocation is ~50% smaller)

If the meta-device approach looks right, happy to clean this up and turn it into a PR. If you would prefer a different solution path (e.g. opt-in flag, smaller injector when Klein checkpoint is loaded onto Dev base, etc.) let me know.

### Related

PR #18 by @konradozog addresses a separate cumulative-monkey-patch bug in the same `apply()` method and is also worth merging.
